### PR TITLE
Feature/eu support

### PR DESCRIPTION
**Pull request by cstryker**

### CODE REVIEW

**Comment by cstryker** [comment]
unfortunately I rebased from develop ... not sure how to exclude those changes from the PR

File: shared/ui/Authentication/Signup.tsx [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { CodeStreamState } from "../store";
 import { FormattedMessage } from "react-intl";
 import Icon from "../Stream/Icon";
 import Button from "../Stream/Button";
+import { InlineMenu } from "../src/components/controls/InlineMenu";
 import { Link } from "../Stream/Link";
+import { ModalRoot } from "../Stream/Modal"; // HACK ALERT: including this component is NOT the right way
```
Comment (bcanzanella) [inline shared/ui/Authentication/Signup.tsx:9]: TODO. fix this.

File: shared/ui/Authentication/SignupNewRelic.tsx [hunk 2]
```diff
@@ -16,7 +16,11 @@ import { logError } from "../logger";
 import { CodeStreamState } from "@codestream/webview/store";
 import { LoginResult } from "@codestream/protocols/api";
 import { goToNewUserEntry, goToCompanyCreation, goToLogin } from "../store/context/actions";
+import { setEnvironment } from "../store/session/actions";
 import { completeSignup } from "./actions";
+import { InlineMenu } from "../src/components/controls/InlineMenu";
+import { ModalRoot } from "../Stream/Modal"; // HACK ALERT: including this component is NOT the right way
```
Comment (bcanzanella) [inline shared/ui/Authentication/SignupNewRelic.tsx:22]: TODO. fix this.

File: shared/ui/store/session/actions.ts [hunk 3]
```diff
@@ -92,6 +92,75 @@ export const switchToTeam = (
...
+export const switchToForeignCompany = (companyId: string) => async (
+	dispatch,
+	getState: () => CodeStreamState
+) => {
+	const { companies, session, users } = getState();
+	const company = companies[companyId];
+	const user = users[session.userId!] as CSMe;
+	const teamId = company.everyoneTeamId;
+	if (!company) {
+		console.error(`Could not switch to organization ${companyId}, not found`);
```
Comment (bcanzanella) [inline shared/ui/store/session/actions.ts:115]: `logError` so it goes to Sentry?

File: shared/ui/store/session/actions.ts [hunk 4]
```diff
@@ -92,6 +92,75 @@ export const switchToTeam = (
...
+		return;
+	} else if (!company.host) {
+		console.error(`Could not switch to organization ${companyId}, not a foregin company`);
```
Comment (bcanzanella) [inline shared/ui/store/session/actions.ts:118]: `logError` so it goes to Sentry?

File: shared/ui/store/session/actions.ts [hunk 5]
```diff
@@ -92,6 +92,75 @@ export const switchToTeam = (
...
+		return;
+	} else if (!company.host.accessToken) {
+		console.error(`Could not switch to organization ${companyId}, no access token`);
```
Comment (bcanzanella) [inline shared/ui/store/session/actions.ts:121]: `logError` so it goes to Sentry?

File: shared/ui/Authentication/Login.tsx [hunk 6]
```diff
@@ -228,9 +236,39 @@ class Login extends React.Component<Props, State> {
...
+			}
+			if (this.props.selectedRegion) {
+				const selectedHost = this.props.environmentHosts.find(
+					host => host.shortName === this.props.selectedRegion
+				);
+				if (selectedHost) {
+					regionSelected = selectedHost.name;
+				}
+			}
+		}
```
Comment (bcanzanella) [inline shared/ui/Authentication/Login.tsx:267]: Wrap this up into a method or component? code like this is in 3 spots.

**Review by bcanzanella — APPROVED** [review]
looks good! i noted the couple spots w/r/t the modal fix.

**Review by marcelofarias — APPROVED** [review]

**Review by eric-j — APPROVED** [review]

File: shared/ui/store/session/actions.ts (hunk 3 above)
Comment (cstryker) [reply shared/ui/store/session/actions.ts:115]: agree

File: shared/ui/store/session/actions.ts (hunk 4 above)
Comment (cstryker) [reply shared/ui/store/session/actions.ts:118]: agree

File: shared/ui/store/session/actions.ts (hunk 5 above)
Comment (cstryker) [reply shared/ui/store/session/actions.ts:121]: agree

File: shared/ui/Authentication/Login.tsx (hunk 6 above)
Comment (cstryker) [reply shared/ui/Authentication/Login.tsx:267]: Well Login.tsx is stuck on class components, while the other two instances in Signup.tsx and NewRelicSignup.tsx are using functional components. IMHO, I think eliminating the duplication between the class component and the functional components is not worth the cost in readability. And even between Signup.tsx and NewRelicSignup.tsx, I don't really think the nature of the code really lends itself to being abstracted out. Neither approach is ideal, but my judgment is to leave as is.

**Pull request closed without merging**